### ISSUE TITLE
Electron 6 Test Failure: Search reports telemetry on search completed

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#80251]
It looks like recently there is a test failure that I can only reproduce from the `electron-6.0.x` branch:

```
1) SearchModel Search Model: Search reports telemetry on search completed:

      AssertionError [ERR_ASSERTION]: false == true
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (file:///Users/bpasero/Development/Microsoft/vscode-electron-6.0.x/out/vs/workbench/contrib/search/test/common/searchModel.test.js:110:20)
      at processTicksAndRejections (internal/process/task_queues.js:89:5)
```

Must have been a regression from not too recent as this is the first time I am seeing that failing.

**Comment by bpasero** [comment]
I pushed https://github.com/microsoft/vscode/commit/e5db2c31e4f7459a084977ed5bafc2a39ff50a37 as workaround. My testing shows that everything seems to be working fine, however:
* when `await testObject.search({ contentPattern: { pattern: 'somestring' }, type: 1, folderQueries }); `returns the `onDoneStopwatch(duration => this.telemetryService.publicLog('searchResultsFinished', { duration }));` has not yet been fired. Assuming that the one happens before the other seems fishy to me.

Maybe something changed with how in E6 async code is ordered. This possibly came in when we switched to compile to ES7 from ES6 and native async await.
